merino/middleware/geolocation.py

```diff
@@ -35,16 +36,18 @@ async def dispatch(self, request: Request, call_next):
         # `request.client.host` should be the first remote client address of `X-Forwarded-For`.
         record = None
         try:
-            record = reader.get(request.client.host)
+            record = reader.city(request.client.host if request.client else "")
```

Comment:
```suggestion
            record = reader.city(request.client.host or "")
```


Per Starlette's doc: `request.client` is always exposed as a namedtuple. But `request.client.host` could be None.

> Client Address
>
> The client's remote address is exposed as a named two-tuple `request.client`. Either item in the tuple may be None.
>
> The hostname or IP address: request.client.host
> The port number from which the client is connecting: request.client.port

Reply:
I tried that at first because i prefer that syntax. however, per the starlette code `request.client` is of type `Optional[Address]` and the definition of `Address` is:

```py
class Address(typing.NamedTuple):
    host: str
    port: int
```

Reply:
Ah, then we have to do `request.client or request.client.host or ""`. Because of that note:

> Either item in the tuple may be None.

Reply:
oh nope, that will return `request.client` in most cases. 😅